The job **`dist-i686-msvc`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/11055720248/job/30715562387) [(plain)](https://github.com/rust-lang-ci/rust/commit/a3b36aad417aa01afcd25e14a8c163bd17f49bd7/checks/30715562387/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Updating files:  98% (49075/49811)
Updating files:  99% (49313/49811)
Updating files: 100% (49811/49811)
Updating files: 100% (49811/49811), done.
branch 'try' set up to track 'origin/try'.
Switched to a new branch 'try'
[command]"C:\Program Files\Git\bin\git.exe" log -1 --format='%H'
'a3b36aad417aa01afcd25e14a8c163bd17f49bd7'
##[group]Run src/ci/scripts/setup-environment.sh
src/ci/scripts/setup-environment.sh
---
file:.git/config remote.origin.url=https://github.com/rust-lang-ci/rust
file:.git/config remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*
file:.git/config gc.auto=0
file:.git/config http.https://github.com/.extraheader=AUTHORIZATION: basic ***
file:.git/config branch.try.remote=origin
file:.git/config branch.try.merge=refs/heads/try
file:.git/config submodule.library/backtrace.url=https://github.com/rust-lang/backtrace-rs.git
file:.git/config submodule.library/stdarch.active=true
file:.git/config submodule.library/stdarch.url=https://github.com/rust-lang/stdarch.git
file:.git/config submodule.src/doc/book.active=true
---
[RUSTC-TIMING] panic_unwind test:false 0.080
[RUSTC-TIMING] std_detect test:false 0.162
[RUSTC-TIMING] hashbrown test:false 0.781
 Documenting std v0.0.0 (C:\a\rust\rust\library\std)
error[E0616]: field `sun_path` of struct `os::unix::net::addr::libc::sockaddr_un` is private
  --> std\src\os\unix\net\addr.rs:23:67
   |
23 | const SUN_PATH_OFFSET: usize = mem::offset_of!(libc::sockaddr_un, sun_path);

For more information about this error, try `rustc --explain E0616`.
error: could not document `std`
warning: build failed, waiting for other jobs to finish...
warning: build failed, waiting for other jobs to finish...
[RUSTC-TIMING] std test:false 4.531
Command has failed. Rerun with -v to see more details.
  local time: Thu, Sep 26, 2024  4:25:22 PM
  network time: Thu, 26 Sep 2024 16:25:22 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/std/src/os/unix/net/addr.rs
+++ b/library/std/src/os/unix/net/addr.rs
@@ -15,7 +15,9 @@ mod libc {
     pub type socklen_t = u32;
     pub struct sockaddr;
     #[derive(Clone)]
-    pub struct sockaddr_un;
+    pub struct sockaddr_un {
+        pub sun_path: [u8; 1],
+    }
 }
 
 const SUN_PATH_OFFSET: usize = mem::offset_of!(libc::sockaddr_un, sun_path);

